avocado/plugins/nlist.py

```diff
@@ -70,7 +71,12 @@ def configure(self, parser):
     def run(self, config):
         references = config.get('nlist.references')
         verbose = config.get('nlist.verbose')
-        resolutions = resolver.resolve(references)
+        hint_filepath = '.avocado.hint'
+        if os.path.exists(hint_filepath):
+            hint = HintParser(hint_filepath)
+            resolutions = hint.get_resolutions()
+        else:
```

Comment:
This kind of conditional doesn't allow one to run a subset of tests while using a hint file. IMO, while a hint file will influence how a reference will be resolved (and turned into `ReferenceResolution` with `Runnables`), one should be able to to:

```
ls tests/
one-test
two-test
```

```sh
avocado nrun tests/one-test
```

And because of:

```ini
[kinds]
tap = ./tests/*test

[tap]
uri = $test
```

It would run `tests/single-test` *only*, even if there are other files that match the content of `tap` (under `[kinds]`).

Reply:
Ok, I think that I understood and agree. Will try to fix this on v2.